src/lib.rs

```diff
@@ -315,49 +374,135 @@ const MAVLINK_IFLAG_SIGNED: u8 = 0x01;
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
 // Follow protocol definition: https://mavlink.io/en/guide/serialization.html#mavlink2_packet_format
-pub struct MAVLinkV2MessageRaw {
-    pub start: u8, // = MAV_STX_V2
-    pub payload_length: u8,
-    pub incompatibility_flags: u8,
-    pub compatibility_flags: u8,
-    pub sequence: u8,
-    pub system_id: u8,
-    pub component_id: u8,
-    pub message_id: u32,
-    pub payload_buffer: [u8; 255],
-    pub checksum: u16,
+pub struct MAVLinkV2MessageRaw([u8; 1 + Self::HEADER_SIZE + 255 + 2 + Self::SIGNATURE_SIZE]);
+
+impl Default for MAVLinkV2MessageRaw {
+    fn default() -> Self {
+        Self::new()
+    }
 }
 
 impl MAVLinkV2MessageRaw {
+    const HEADER_SIZE: usize = 9;
+    const SIGNATURE_SIZE: usize = 13;
+
+    pub const fn new() -> Self {
+        Self([0; 1 + Self::HEADER_SIZE + 255 + 2 + Self::SIGNATURE_SIZE])
+    }
+
+    #[inline]
+    pub fn header(&mut self) -> &[u8] {
+        &self.0[1..=Self::HEADER_SIZE]
+    }
+
+    #[inline]
+    fn mut_header(&mut self) -> &mut [u8] {
+        &mut self.0[1..=Self::HEADER_SIZE]
+    }
+
+    #[inline]
+    pub fn payload_length(&self) -> u8 {
+        self.0[1]
+    }
+
+    #[inline]
+    pub fn incompatibility_flags(&self) -> u8 {
+        self.0[2]
+    }
+
+    #[inline]
+    pub fn compatibility_flags(&self) -> u8 {
+        self.0[3]
+    }
+
+    #[inline]
+    pub fn sequence(&self) -> u8 {
+        self.0[4]
+    }
+
+    #[inline]
+    pub fn system_id(&self) -> u8 {
+        self.0[5]
+    }
+
+    #[inline]
+    pub fn component_id(&self) -> u8 {
+        self.0[6]
+    }
+
+    #[inline]
+    pub fn message_id(&self) -> u32 {
+        u32::from_le_bytes([self.0[7], self.0[8], self.0[9], 0])
+    }
+
+    #[inline]
     pub fn payload(&self) -> &[u8] {
-        &self.payload_buffer[..self.payload_length as usize]
+        let payload_length = self.payload_length() as usize;
+        &self.0[(1 + Self::HEADER_SIZE)..(1 + Self::HEADER_SIZE + payload_length)]
     }
 
-    pub fn mut_payload(&mut self) -> &mut [u8] {
-        &mut self.payload_buffer[..self.payload_length as usize]
+    #[inline]
+    pub fn checksum(&self) -> u16 {
+        let payload_length = self.payload_length() as usize;
+        u16::from_le_bytes([
+            self.0[1 + Self::HEADER_SIZE + payload_length],
+            self.0[1 + Self::HEADER_SIZE + payload_length + 1],
+        ])
+    }
+
+    fn mut_payload_and_checksum_and_sign(&mut self) -> &mut [u8] {
+        let payload_length = self.payload_length() as usize;
```

Comment:
use lossless conversion

i won't bother writing all of these out. You should be find them all with something like

`cargo +nightly clippy -- -A clippy::all -W clippy::cast_lossless`